### ISSUE TITLE
ipynb files not adhering to cspell.json

### ISSUE THREAD

**Issue by RobWatersMet** [streetsidesoftware/vscode-spell-checker#3787]
Here is a simple example: 

If I add the word xarray to cspell.json - everything works fine for .md and .py files:

![image](https://github.com/user-attachments/assets/6b4b1782-2270-4ff4-9a7f-019d5970e193)

But inside .ipynb files the cspell.json addition are not working, despite cspell running

![image](https://github.com/user-attachments/assets/f2deb61e-d55c-4891-8211-02cba6058e0b)

If I add the words to settings.json for the workspace:

 {
    "cSpell.words": [
        "xarray"
    ]
}

it works! 

But ideally I'd like the whole workspace, including .ipynb, to use the cspell.json which is tracked. 

I am running vscode on wsl2, Ubuntu-22.04.

P.S. the extension is freaking great and helps me a lot - thanks :)

**Comment by RobWatersMet** [comment]
if its something I'm mistakenly doing, let me know

**Comment by Jason3S** [comment]
@RobWatersMet,

What version of the spell checker are you using?

This is what I see:

<img width="443" alt="image" src="https://github.com/user-attachments/assets/6155145a-ad22-4203-ab03-8c2c1e81f225">

<img width="552" alt="image" src="https://github.com/user-attachments/assets/ff4e384a-7043-4784-b073-0f30ab475221">
<img width="392" alt="image" src="https://github.com/user-attachments/assets/410891ad-055a-42f9-91a5-5e39d6d3dc33">

**Comment by RobWatersMet** [comment]
It's now not working at all. I think this is the exact same setup and I've just tried 4.0.16, which is what I would have been using when I first posted. 

```
fetchSettingsForUri: file://wsl%2Bubuntu-22.04/home/rw796/code/dummy_workspace/test.ipynb TypeError: Invalid URL
    at new URL (node:internal/url:806:29)
    at sp (/home/rw796/.vscode-server/extensions/streetsidesoftware.code-spell-checker-4.0.16/packages/_server/dist/main.cjs:223:13843)
    at rK (/home/rw796/.vscode-server/extensions/streetsidesoftware.code-spell-checker-4.0.16/packages/_server/dist/main.cjs:223:13190)
    at nK (/home/rw796/.vscode-server/extensions/streetsidesoftware.code-spell-checker-4.0.16/packages/_server/dist/main.cjs:223:13291)
    at l0.rootSchemaAndDomainForUri (/home/rw796/.vscode-server/extensions/streetsidesoftware.code-spell-checker-4.0.16/packages/_server/dist/main.cjs:223:20779)
    at l0.rootSchemaAndDomainFolderForUri (/home/rw796/.vscode-server/extensions/streetsidesoftware.code-spell-checker-4.0.16/packages/_server/dist/main.cjs:223:20939)
    at l0.__fetchSettingsForUri (/home/rw796/.vscode-server/extensions/streetsidesoftware.code-spell-checker-4.0.16/packages/_server/dist/main.cjs:223:23147)
    at async /home/rw796/.vscode-server/extensions/streetsidesoftware.code-spell-checker-4.0.16/packages/_server/dist/main.cjs:223:22261
    at async l0._fetchSettingsForUri (/home/rw796/.vscode-server/extensions/streetsidesoftware.code-spell-checker-4.0.16/packages/_server/dist/main.cjs:223:22202) {
  code: 'ERR_INVALID_URL',
  input: 'file://wsl%2Bubuntu-22.04/home/rw796/code/dummy_workspace/test.ipynb'
}
```

**Comment by Jason3S** [comment]
@RobWatersMet,

Thank you. I'm guessing you are running on windows.

Can you set the log level to debug:
<img width="621" alt="image" src="https://github.com/user-attachments/assets/eaaab864-1165-4db1-ab37-d11e89e481ad">

Then open the console output, select `Code Spell Checker Server`
<img width="1303" alt="image" src="https://github.com/user-attachments/assets/e1c3e1c3-e1f2-42b8-9ece-9c26ba3c9bc6">

Then open `test.ipynb`.

Can you paste the line that says `getActiveUriSettings`, like this:
```
316	2024-11-18T19:07:13.267Z	getActiveUriSettings	vscode-notebook-cell:/Users/jason/projects/vscode-spell-checker/fixtures/workspaces/jupyter/getting-started/sample.ipynb#W1sZmlsZQ%3D%3D
```

The issue is that it is failing to covert the `vscode-notebook-cell:` url into a `file:` url.

**Comment by Jason3S** [comment]
@RobWatersMet,

The cause of the issue is related to the URI Notebooks use. They use their own custom scheme. It generally is of the form: 
```
vscode-notebook-cell:/Users/jason/projects/vscode-spell-checker/fixtures/workspaces/jupyter/getting-started/sample.ipynb#W2sZmlsZQ%3D%3D
```

The spell checker tries to convert it into a `file:` URL. But JavaScript doesn't like it and throws an error.

The Spell Checker log should show the URI.

**Comment by RobWatersMet** [comment]
I can confirm https://github.com/streetsidesoftware/vscode-spell-checker/pull/3826 my issue

Thanks for the rapid update - much appreciated

**Comment by atticusmatticus** [comment]
I also can confirm that this fixes my issue #3819 

thank you!
